llama_index/pipeline.py

```diff
@@ -0,0 +1,42 @@
+# from functools import
+import json
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field, asdict
+from typing import Dict, List, Any
+
+
+@dataclass
+class PipelineSchema:
+    """Class representing a component in a pipeline."""
+
+    name: str
+    metadata: Dict[str, Any] = field(default_factory=dict)
+    children: List["PipelineSchema"] = field(default_factory=list)
```

Comment:
Perhaps we need to distinguish between children and inputs.

Children are executed as parts of the overall pipeline.

Inputs have their outputs passed into the pipeline.

Reply:
Actual usage will be handled in part 2/2